api/base_controller.py

```diff
@@ -82,6 +86,38 @@ def authenticate(self):
         data = self.manager.authentication_for_opds_document
         return Response(data, 401, headers)
 
+    def library_through_external_loan_identifier(self, loan_external_identifier):
+        """Look up the library the user is trying to access using a loan's external identifier.
+        We assume that the external identifier is globally unique which is true, for example,
+        in the case of using Readium LCP.
+
+        :param loan_external_identifier: External identifier of the patron's loan
+        :type loan_external_identifier: basestring
+
+        :return: Library the patron is trying to access
+        :rtype: Library
+        """
+        self.manager.reload_settings_if_changed()
+
+        loan = None
+
+        try:
+            loan = get_one(self._db, Loan, external_identifier=loan_external_identifier)
+        except:
+            logging.exception(
+                'An unexpected exception occurred during fetching a loan with external identifier {0}'.format(
+                    loan_external_identifier
+                )
+            )
```

Comment:
This method should either re-throw the exception or return a problem detail document; probably the former since an exception here is most likely a problem with the database.

Reply:
[Line # 113](https://github.com/NYPL-Simplified/circulation/pull/1495/files/2ab30d75eb96d2ec2fe9b40f6032263fd5c446e3#diff-f258e5927d0cab67f7bc9611720642a5R113) checks whether `loan` is None and returns `LOAN_NOT_FOUND` if it's empty.
Do you want to return a more specific problem detail document in the case of an exception?

Reply:
Normally we wouldn't catch this exception at all -- any database access might raise an unexpected exception. If that happens we would allow it to be treated as an uncaught exception because there's a serious problem with the database. My preferred solution would be to get rid of the try/except block altogether, but if you saw something unusual during development, returning a specific problem detail might be appropriate.

Reply:
Removed try/except block